tests/test_load_data.py

```diff
@@ -27,271 +37,239 @@ class TestLoadData(SMRFTestCase):
+        config.raw_cfg['time']['end_date'] = '1900-02-01 00:00'
+
+        config.apply_recipes()
+        config = cast_all_variables(config, config.mcfg)
+
+        with self.assertRaises(Exception):
+            result = run_smrf(config)
+
+class TestLoadCSVData(SMRFTestCase):
+
+    def test_station_dates(self):
+        """
+        Test the start date not in the data
+        """
+
+        # Test the start date
+        config = deepcopy(self.base_config)
+
+        # Use dates not in the dataset, expecting an error
+        config.raw_cfg['time']['start_date'] = '1900-01-01 00:00'
+        config.raw_cfg['time']['end_date'] = '1900-02-01 00:00'
+
+        # apply the new recipies
+        config.apply_recipes()
+        config = cast_all_variables(config, config.mcfg)
+
+        result = self.can_i_run_smrf(config)
+
+        # test the base run with the config file
+        self.assertFalse(result)
+
+
+    def test_all_stations(self):
+        """
+        Test using all stations
+        """
+
+        # test the end date
+        config = deepcopy(self.base_config)
+        config.raw_cfg['csv']['stations'] = ['RMESP', 'RME_176']
 
+        # apply the new recipies
+        config.apply_recipes()
+        config = cast_all_variables(config, config.mcfg)
+
+        # test the base run with the config file
+        result = self.can_i_run_smrf(config)
+        assert result
+
+
+class TestLoadGrid(SMRFTestCase):
 
     def test_grid_wrf(self):
         """ WRF NetCDF loading """
 
         config = deepcopy(self.base_config)
         del config.raw_cfg['csv']
 
-        wrf_grid = {'data_type': 'wrf',
-                    'file': './RME/gridded/WRF_test.nc',
-                    'zone_number': 11,
-                    'zone_letter': 'N'}
-        config.raw_cfg['gridded'] = wrf_grid
-#         config.raw_cfg['system']['max_values'] = 2
-        config.raw_cfg['system']['threading'] = False
-#         config.raw_cfg['system']['timeout'] = 10
+        adj_config = {'gridded':{'data_type': 'wrf',
+                                 'wrf_file': './RME/gridded/WRF_test.nc',
+                                 'zone_number': '11',
+                                 'zone_letter': 'N'},
+                      'system':{'threading':'False',
+                                'log_file': './output/log.txt'},
+                     'precip':{'station_adjust_for_undercatch': 'False'},
+                     'thermal':{'correct_cloud': 'False',
+                                'correct_veg': 'True'},
+                     'time':{'start_date':'2015-03-03 00:00',
+                             'end_date': '2015-03-03 04:00'}}
 
-        # set the distrition to grid, thermal defaults will be fine
-        variables = ['air_temp', 'vapor_pressure', 'wind', 'precip', 'solar', 'thermal']
-        for v in variables:
-             config.raw_cfg[v]['mask'] = False
+        config.raw_cfg.update(adj_config)
```

Comment:
Does the config update also redo the recipes if they changed?

Reply:
No. The update is a dictionary func for combining dictionaries. I do rerun ucfg.apply_recipes() and cast_all_variables(ucfg) after we make these mods to ensure we are populating everything.